src/jbig2.cc

```diff
@@ -353,6 +355,24 @@ main(int argc, char **argv) {
       continue;
     }
 
+    if (strcmp(argv[i], "-D") == 0 ||
+        strcmp(argv[i], "--dpi") == 0) {
+      char *endptr;
+      long t_dpi = strtol(argv[i+1], &endptr, 10);
+      if (*endptr) {
```

Comment:
It looks like the indentation of this section got screwed up, but the patch looks nice (not yet tested), since it would avoid having waste time (and space) to convert bazillion pages to an intermediary format just to set the DPIs (and let leptonica decode all the images that were coded into the intermediate format).